src/components/theme-selector/theme-selector.element.ts

```diff
@@ -1,36 +1,38 @@
-import { html, LitElement, TemplateResult } from 'lit'
+import { html, TemplateResult } from 'lit'
 import { customElement, state } from 'lit/decorators'
-import { Subscription } from 'rxjs'
-import { ThemeNodeModel } from '../../../services/themes/themes.model'
-import { themesService } from '../../../services/themes/themes.service'
-import { themingService } from './theme/theming.service'
+import { ThemeNodeModel } from '../../services/themes/themes.model'
+import { themesService } from '../../services/themes/themes.service'
+import { themeSelectorService } from './theme-selector.service'
 
-import './theme/theme-grid.component'
-import './theme/theme-selector-button.component'
+import './theme-grid.element'
+import './theme-selector-button.element'
+import { map } from 'rxjs'
+import LuxElement from '../common/base.element'
 
 @customElement('lux-theme-selector')
-export class ThemeSelector extends LitElement {
+export class ThemeSelectorElement extends LuxElement {
   @state()
   private isOpen = false
   @state()
   private currentTheme?: ThemeNodeModel
   @state()
   private themes?: ThemeNodeModel[]
-  private subscription: Subscription
+
   constructor() {
     super()
-    this.subscription = themesService.theme$.subscribe(theme => {
-      if (theme) {
-        this.currentTheme = theme
-        themingService.setCurrentThemeColors(theme)
-      }
-    })
-    this.subscription.add(
-      themesService.themes$.subscribe(themes => {
-        this.themes = themes.filter(
-          theme => theme.metadata?.display_in_switcher === true
+
+    this.subscribe(
+      'currentTheme' as keyof this,
+      themesService.theme$
+    ).subscribe(theme => themeSelectorService.setCurrentThemeColors(theme))
```

Comment:
Wondering if the second `subscribe` here would be unsubcribed. Maybe
```suggestion
    this.subscribe(
      'currentTheme' as keyof this,
      themesService.theme$.pipe(
        map(theme => themeSelectorService.setCurrentThemeColors(theme))
      )
    )
```
would be an alternative ?

Reply:
No, sorry, this won't update `currentTheme`